### ISSUE TITLE
ENH: Pandas Tensor Data Type

### ISSUE THREAD

**Issue by bionicles** [pandas-dev/pandas#59006]
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

Reviewing Arrow docs link from @WillAyd, spotted this 

https://arrow.apache.org/docs/format/CanonicalExtensions.html#variable-shape-tensor 

Tensor is _exactly_ what I'm talking about in Additional Context [1] and would enable Pandas users to have a column datatype for big blocks of some underlying type

### Feature Description

Support Arrow Tensor in Pandas 

Python
https://arrow.apache.org/docs/python/generated/pyarrow.Tensor.html#pyarrow.Tensor

Rust
https://github.com/apache/arrow-rs/blob/3715d5447e468a5a4dc631ae9aafec706c57aa20/arrow/src/tensor.rs#L115 

### Alternative Solutions

just make everything an "object":
```
>>> import numpy as np
>>> import pandas as pd
>>> x = {'hello': 'world'}
>>> y = np.ones(3)
>>> df = pd.DataFrame({'X': [x], 'Y': [y]})
>>> df
                    X                Y
0  {'hello': 'world'}  [1.0, 1.0, 1.0]
>>> df.dtypes
X    object
Y    object
dtype: object
```

### Additional Context

[1] https://github.com/pandas-dev/pandas/pull/58455#issuecomment-2161603939 onward

**Comment by mroeschke** [comment]
cc @jbrockmendel if pyarrow plans to support it's compute functions for `pyarrow.Tensors`, this may be the appropriate 2D EA block backing for `ArrowExtensionArray`

**Comment by WillAyd** [comment]
I think the nullability bitmap for the extension array only applies to the entire datum itself, not to individual records within each struct

**Comment by jbrockmendel** [comment]
I don’t think this belongs in pandas.

**Comment by bionicles** [comment]
Hi, I thought a lot of data is / are describable as tensors, and having a nice way to have "a column of 3D videos" or "a column of 2D images in RGBA format" etc might make it more useful for common machine learning and data science use cases, that's why I made the request

could align with the sql standard which added n dimensional arrays in 2019 and with Arrow which has a tensor type

Could having columns of n-dimensional arrays enable a richer language to describe data?

> pandas aims to be the fundamental high-level building block for doing practical, real world data analysis in Python. Additionally, it has the broader goal of becoming the most powerful and flexible open source data analysis / manipulation tool available in any language.

**Comment by jbrockmendel** [comment]
its a fine use case for a 3rd party EA, just not one i think we should be implementing ourselves.  i guess if pyarrow supports it then `pd.ArrowDtype(pa.tensor_whatever())` could work.

**Comment by mroeschke** [comment]
Agreed with others that this probably belongs as a 3rd party EA so closing
